src/handlers/sdp/RemoteSdp.ts

```diff
@@ -172,6 +174,24 @@ export class RemoteSdp {
 			codecOptions,
 		});
 
+		const mediaObject = mediaSection.getObject();
+
+		// Remove Dependency Descriptor extension unless there is support for
+		// the codec in mediasoup.
+		// @ts-expect-error --- mediaObject type is 'any'.
+		const ddCodec = mediaObject.rtp.find((rtp: any) =>
+			DD_CODECS.includes(rtp.codec.toLowerCase())
+		);
+
+		if (!ddCodec) {
+			// @ts-expect-error --- mediaObject type is 'any'.
+			mediaObject.ext = mediaObject.ext.filter(
+				(extmap: any) =>
+					extmap.uri !==
+					'https://aomediacodec.github.io/av1-rtp-spec/#dependency-descriptor-rtp-header-extension'
+			);
+		}
```

Comment:
My initial expectation was that this should be done by calling some útil function from each handler but now I understand that this is not related to each handler but to mediasoup server which only supports DD for AV1. 

So despite I don't like that this is hardcoded in client I understand that we don't signal anything like "only AV1 supports DD" from mediasoup to client since that info can not be expressed via Router capabilities. So I think this is indeed the best we can do for now.